starknet_py/net/schemas/gateway.py

```diff
@@ -116,7 +116,7 @@ def _deserialize(self, value, attr, data, **kwargs):
 
 
 class SierraCompiledContractSchema(SierraContractClassSchema):
-    abi = AbiField(data_key="abi", required=True)
+    raw_abi = AbiField(data_key="abi", required=True)
```

Comment:
I believe there is not reason to have `AbiField` here instead of `fields.String`, but please double check.

Reply:
no idea why, but I get `marshmallow.exceptions.ValidationError: {'abi': ['Not a valid string.']}`

Reply:
Is there any specific test that throws this error?

Reply:
![Screenshot 2024-03-08 at 09 15 51](https://github.com/software-mansion/starknet.py/assets/15279784/e3bb3c6a-3853-4874-aa77-85b622b04dbe)
Many of them

Reply:
![Screenshot 2024-03-08 at 09 29 12](https://github.com/software-mansion/starknet.py/assets/15279784/0552ba16-84e2-41e7-a562-26b8eb0c6cee)

Reply:
tip: let's not use screenshots but code blocks instead. makes it way easier for a reviewer to find the places you're reffering to in code.

Reply:
I investigated it, and my conclusion is: SierraContractClassSchema we use to map [model from RPC](https://github.com/starkware-libs/starknet-specs/blob/eedf5f899aa51a85a841333175023aa5d615aa33/api/starknet_api_openrpc.json#L3240), so abi is string. SierraCompiledContractSchema we use to load compiled contract from a file, so abi is Dict (maybe str)